go-controller/pkg/util/net.go

```diff
@@ -29,6 +30,52 @@ func intToIP(i *big.Int) net.IP {
 	return net.IP(i.Bytes())
 }
 
+// IPToUint32 returns a uint32 of an IPv4/IPv6 string
+func IPToUint32(egressIP string) uint32 {
+	ip := net.ParseIP(egressIP)
+	if utilnet.IsIPv6(ip) {
+		// This can obviously not be done for IPv6. But the logic here is:
+		// "allow users to create IPv6 egress IP addresses with a 1/(2^32)
+		// probability that they might collide. Or just use shared gateway
+		// mode, and live without this risk."
+		return binary.BigEndian.Uint32(ip[12:16])
+	}
+	return binary.BigEndian.Uint32(ip)
+}
+
+// GetNodeGatewayRouterAddress returns the IPs (IPv4 and/or IPv6) of the provided node's gateway router
+func GetNodeGatewayRouterAddress(nodeName string) (net.IP, net.IP, error) {
+	stdout, _, err := RunOVNNbctl(
+		"--data=bare",
+		"--format=table",
+		"--no-heading",
+		"--columns=options",
+		"find", "logical_router",
+		fmt.Sprintf("name=GR_%s", nodeName),
+		"options:lb_force_snat_ip!=-",
```

Comment:
I think this is cleaner:
[root@ovn-control-plane ~]# ovn-nbctl --data=bare --format=json --columns networks find logical_router_port name=rtoe-GR_ovn-worker
{"data":[["169.254.33.2/24"]],"headings":["networks"]}

Then just json.unmarshal the data.

Reply:
I will handle this in a later patch as I will be off the coming two days and don't want to hold off this PR because of this.